dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateVectorStore.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using Microsoft.SemanticKernel.Data;
+using Microsoft.SemanticKernel.Http;
+
+namespace Microsoft.SemanticKernel.Connectors.Weaviate;
+
+/// <summary>
+/// Class for accessing the list of collections in a Weaviate vector store.
+/// </summary>
+/// <remarks>
+/// This class can be used with collections of any schema type, but requires you to provide schema information when getting a collection.
+/// </remarks>
+public sealed class WeaviateVectorStore : IVectorStore
+{
+    /// <summary><see cref="HttpClient"/> that can be used to manage the collections in Weaviate.</summary>
```

Comment:
Super nit :) The http client is not used for collection management, right? It's for accessing the store API?

`/// <summary>An instance of <see cref="HttpClient"/> used to interact with Weaviate's API.</summary>`

Reply:
It's for accessing the store API for collection management :) But I've updated the documentation to better explain the purpose of `HttpClient` in this class.